src/main/java/com/microsoft/graph/models/extensions/Multipart.java

```diff
@@ -33,46 +38,132 @@ public Multipart() {
 	 * Get the multipart boundary for use in the request header
 	 * @return the multipart boundary
 	 */
-	public String boundary() {
+	public String getBoundary() {
 		return boundary;
 	}
 	
+	/**
+	 * Set the multipart boundary for use in the request header
+	 * @param boundary The multipart boundary
+	 */
+	public void setBoundary(String boundary) {
+		this.boundary = boundary;
+	}
+	
+	/**
+	 * Get the contentType for use in the request header
+	 * @return the multipart Content-Type
+	 */
+	public String getContentType() {
+		return contentType;
+	}
+	
+	/**
+	 * Set the contentType for use in the request header
+	 * @param contentType The multipart Content-Type
+	 */
+	public void setContentType(String contentType) {
+		this.contentType = contentType;
+	}
+	
 	/**
 	 * Get the Content-Type header to send the multipart request
 	 * @return the multipart header option
 	 */
 	public HeaderOption header() {
-		return new HeaderOption("Content-Type", "multipart/form-data; boundary=\"" + boundary + "\"");
+		return new HeaderOption("Content-Type", contentType + "; boundary=\"" + boundary + "\"");
+	}
+	
+	private void writePartData(String partContent, byte[] byteArray) throws IOException{
+		out.write(partContent.getBytes(MULTIPART_ENCODING));
+		out.write(byteArray);
+		String returnContent = RETURN + RETURN;
+		out.write(returnContent.getBytes(MULTIPART_ENCODING));
+	}
+	
+	private String createPartHeader(Map<String, String> headers, String name, String contentType, String filename) {
+		String partContent = addBoundary();
+		if(headers != null) {
+			List<String> listContentDisposition = Arrays.asList("filename","creation-date","modification-date","read-date",
+					"size","name","voice","handling","preview-type");			
+			List<String> mainHeaders = Arrays.asList("Content-Disposition","Content-Type","charset");
+			
+			if(headers.containsKey("Content-Disposition")) {			
+				partContent += "Content-Disposition:"+headers.get("Content-Disposition")+";"; 
+				for (Map.Entry<String,String> entry : headers.entrySet()) {
+					if(listContentDisposition.contains(entry.getKey())) {
+						partContent += " " + entry.getKey() + "=\"" + entry.getValue() + "\";";
+					}
+				}
+				partContent = partContent.substring(0, partContent.length()-1);
```

Comment:
You want to strip the final semicolon. Why's that?

Reply:
Content-Disposition parameters are prefixed by semi-colon, so removing the trailing one is technically correct.

However, I'm a little puzzled as to this approach where we pretend Content-Disposition parameters are headers and then we lift them out of the headers collection to construct the Content-Disposition header value.  The list of headers should just be a list of HTTP headers where the content-disposition value is pre-constructed.  If we need a helper to build that value, then let's build a createContentDispositionValue() function.

Reply:
- I have now changed the way in developer will use this addPart(Map<String, String> headers, byte[] content) and added Helper function String createContentHeaderValue(String contentValue, Map<String, String> contentDispParameter). 
- contentDispParameter map will contain (key=parameterName, Value=parameterValue). eg. (key="name", value="name-val").
- createContentHeaderValue helper function will construct the string "contentValue;name="nameval";..." which will be then passed in headers map as (key=Content-Disposition, value=string from helper function), (key=Content-Type, value=string from helper).